src/Particle/DistanceTable.h

```diff
@@ -31,10 +31,10 @@ class ResourceCollection;
 /** @ingroup nnlist
  * @brief Abstract class to manage pair data between two ParticleSets.
  *
- * Each DistanceTableData object is fined by Source and Target of ParticleSet types.
+ * Each DistanceTable object is fined by Source and Target of ParticleSet types.
  *
  */
-class DistanceTableData
+class DistanceTable
```

Comment:
I think it would be better if this was just an interface i.e. a pure virtual base class. Push the data members into the derived classes as well as all the accessors, which I believe are always called with knowledge of the type.

Reply:
Isn't this exactly what I did in this PR? I added more comments above to make it clear ;)

Reply:
That's what I thought, I just wanted to push it the rest of the way.